---
title: "DEG Analysis"
author: "Andrew D. Johnston"
date: "03/20/2018"
output:
  html_document:
  css: styles.css
toc: yes
toc_depth: 4
pdf_document:
  toc: yes
word_document: default
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(cache=TRUE)
```

```{r packages, echo=TRUE, warning=FALSE, message = FALSE}
# setwd("/Volumes/home/greally-lab/Claudia_Andrew/CRISPR_Proj_combined")

# load in libraries 
library(DESeq2)
library(EDASeq)
library(matrixStats)
library(RUVSeq)
library(qvalue)
library(genefilter)
library(RColorBrewer)
library(pheatmap)
library(UpSetR)
library(RFmarkerDetector)
library(ggplot2)
library(ggthemes)
library(VennDiagram)
library(GeneOverlap)

# set options
options(scipen=999, stringsAsFactors = FALSE)
```

The goal of this analysis is to 1) assess the robust activation of CD34 upon transient 
transfection of HEK 293T cells with dCas9-VP160 and 2) to assess the potential off-target effects of this method. We perform a simple differential expression analysis among our three treatments_r1: control, tranfection with dCas9-VP160 w/o gRNAs, and transfection with dCas9-VP160 and gRNAs targeting the promoter of CD34. We perform differential gene expression analysis incorporating two replicate experiments (each with two replicates per treatment). 

The RNA-seq libraries were prepared similarly with ERCC spike-ins. However, the spike-ins could not be used to normalize the libraries due to their low count number; this most likely is due to the spike-ins not being added at a high enough concentration. Another important caveat to the libraries is that in the first replicate the control library was generated from cells were passed through the FACS machine, whereas in the second experiment set, the control cells were not passed through the FACS machine. For our analysis, we treated the controls as the same. 

First we read in the two experiments (and their two technical replicates for each treatment).
```{r read-in, echo=TRUE, warning=FALSE, message = FALSE}
# reading in and merging the counts

## selecting the files to read in
files <- grep(pattern = "ReadsPerGene.out.tab", x = list.files(), value = TRUE)

files

list_counts <- list()
for (i in 1:length(files)){
  list_counts[[i]] <- read.table(paste(files[i]))
  if (i < 2) {
    df_counts <- list_counts[[1]][,c(1,4)]
  }
  else {
    df_counts <- merge(df_counts, list_counts[[i]][,c(1,4)], by = "V1")
  }
}
dim(df_counts) #  60700    13

## remove the ambiguous, multimapp, no feature, and unmapped read totals
df_counts <- df_counts[-c(60697:60700),]
rownames(df_counts) <- df_counts[,1]
df_counts <- df_counts[,-1]

colnames(df_counts) <- c("Ctrl_1_1", "Ctrl_1_2", "CRISPR_1_1", "CRISPR_1_2", "CD34_1_1",
                         "CD34_1_2", "CD34_2_1", "CD34_2_2", "CRISPR_2_1","CRISPR_2_2",
                         "Ctrl_2_1", "Ctrl_2_2")
head(df_counts)
```

Next we filter the RNAs to be analzyed. First, we apply a simple filter for only 
those RNAs that are expressed at high levels. The RNA must have at least 5 counts 
in four of the samples, thus allowing only genes expressed by only one treatment
group to be retained. Next, we filter for protein coding genes only or protein
coding and long non-coding RNAs.

```{r combine-filter, echo=TRUE, warning=FALSE, message = FALSE}
# expression filter
idx_filt_exp_com <- apply(df_counts, 1, function(x) length(x[x>5])>=4) 
head(idx_filt_exp_com)
filtered_com <- df_counts[idx_filt_exp_com,]
dim(filtered_com) # 19,244     12

# remove spike ins
spikes_com <- grep("ERCC", rownames(filtered_com))
length(spikes_com) # 12
filterd_noSpike_com <- filtered_com[-spikes_com,]
dim(filterd_noSpike_com) # 19232 12

# filter for only protein coding RNAs
prot_ensg_ID <- read.table("../../indexes/Hg38_rel79_ERCC/prot_ENSG.txt")
dim(prot_ensg_ID) # 22002     1
filterd_noSpike_pc_com <- filterd_noSpike_com[
  rownames(filterd_noSpike_com) %in% prot_ensg_ID$V1,]
dim(filterd_noSpike_pc_com) # 14260    6

```

Let's look at the PCA and RLE plots.

```{r combine-EDA, echo=TRUE, warning=FALSE, message = FALSE}

# resorting the columns so that controls, CRISPRs, and CD34s are next to each other
filterd_noSpike_pc_com <- filterd_noSpike_pc_com[,c(1,2,11,12,3,4,9,10,5,6,7,8)]

# set a factor for different treatments
treatments_com <- as.factor(rep(c("Ctrl","CRISPR","CD34"), each=4)) 
treatments_com <- relevel(treatments_com, c("Ctrl"))
replicates_com <- as.factor(rep(c("Rep1", "Rep2", "Rep1", "Rep2", "Rep1", "Rep2"),each=2))

# create expression sets 
eset_pc_com <- newSeqExpressionSet(as.matrix(filterd_noSpike_pc_com),
                                  phenoData = data.frame(treatments_com, 
                                                   row.names=colnames(filterd_noSpike_pc_com)))

# choose a color set
colors_com <- brewer.pal(6, "Dark2")
colors <- brewer.pal(3, "Dark2")


# Make RLE plots
plotRLE(eset_pc_com, outline=FALSE, ylim=c(-4, 4), col=colors[treatments_com],
        main="Protein coding RNAs before normalization") 
limma::plotMDS(counts(eset_pc_com), dim=c(2,3))

# Make PCA plots
plotPCA(eset_pc_com, col=colors[treatments_com], cex=1.2, 
        main = "Protein coding RNAs before normalization")
plotPCA(eset_pc_com, k=3, col=colors[treatments_com], cex=1.2, 
        main="Protein coding RNAs before normalization") 
plotPCA(eset_pc_com, k=3, col=colors[replicates_com], cex=1.2, 
        main="Protein coding RNAs before normalization") 

```

The different treatment groups cluster together as expected except for CRISPR_1_2. Also PC3 seems to be driven by replicate (1v2). 


Next we normalize based on housekeeping gene expression. House keeping genes were
identified by a previous study "Human housekeeping genes revisited" E. Eisenberg
and E.Y. Levanon, Trends in Genetics, 29 (2013) and a list is avaialble for 
download at (https://www.tau.ac.il/~elieis/HKG/). We took only the bottom quartile 
variance house keeping genes to use for normalization.
```{r combine-normalize, echo=TRUE, warning=FALSE, message = FALSE}

# read in house keeping genes 
HK_genes <- read.table("HK_ensembl_ID.txt")
dim(HK_genes) # 4202    1

# grab the HK genes from RNAs being analyzed 
HK_pc_com <- filterd_noSpike_pc_com[which(rownames(filterd_noSpike_pc_com) %in% HK_genes[,1]),]
dim(HK_pc_com) # 3753    12

# examine the variance of the HK genes and take only the bottom 1000 genes to normalize with
## for protein coding RNAs only
HK_pc_com_rsd <- apply(as.matrix(HK_pc_com), 1, rsd)
boxplot(HK_pc_com_rsd)
summary(HK_pc_com_rsd)
HK_pc_lowRSD <- sort(HK_pc_com_rsd)[1:1000]

# Normalize using the house keeping genes
eset_pc_norm_com <- RUVg(eset_pc_com, names(HK_pc_lowRSD), k=1) 

# The weights have been added to the phenotype data
pData(eset_pc_norm_com)

# Make RLE plots
plotRLE(eset_pc_norm_com, outline=FALSE, ylim=c(-4, 4), col=colors[treatments_com],
        main="Protein coding RNAs after normalization") 

# Make PCA plots
plotPCA(eset_pc_norm_com, col=colors[treatments_com], cex=1.2, 
        main = "Protein coding RNAs after normalization")
plotPCA(eset_pc_norm_com, k=3, col=colors[treatments_com], cex=1.2, 
        main="Protein coding RNAs after normalization") 

```

Now PC1 captures the differences between the transfected and non-transfected treatments. PC2 is now based upon replicate. PC3 captures the differences among the treatments perfectly. We will continue with this since the replicate will be included in the DEseq model. 

Next, we perform the differential expression among the different treatments
```{r combine-DEseq, echo=TRUE, warning=FALSE, message = FALSE}

#adding the replicates information to the pData
pData(eset_pc_norm_com) <- cbind(pData(eset_pc_norm_com), replicates_com)

# convert the expression set to a DESeq object
dds_pc_com <- DESeqDataSetFromMatrix(countData = counts(eset_pc_norm_com), 
                                    colData = pData(eset_pc_norm_com), 
                                    design = ~W_1 + replicates_com + treatments_com)

# Run DESeq Wald tests
dds_pc_com <- DESeq(dds_pc_com)

# generate results among the different treatments_com and set a log fold change threshold of 1
ruv_res_con_CD34_pc_com <- results(dds_pc_com, lfcThreshold=1, altHypothesis="greaterAbs", 
                       contrast = c("treatments_com", "CD34", "Ctrl"), alpha=0.05)

ruv_res_con_crisp_pc_com <- results(dds_pc_com, lfcThreshold=1, altHypothesis="greaterAbs",
                             contrast = c("treatments_com", "CRISPR", "Ctrl"), alpha=0.05)

ruv_res_crisp_CD34_pc_com <- results(dds_pc_com, lfcThreshold=1, altHypothesis="greaterAbs", 
                                 contrast = c("treatments_com", "CD34", "CRISPR"), alpha=0.05)

# draw MA plots of results 
## draw horizontal lines for log fold change threshold
drawLines <- function() abline(h=c(-1,1),col="dodgerblue",lwd=2)
ylim<-c(-8,8)

##draw the MA plots
DESeq2::plotMA(ruv_res_con_CD34_pc_com, 
               main="RUV Ctrl vs CD34 PC-com", ylim=ylim); drawLines()
DESeq2::plotMA(ruv_res_con_crisp_pc_com, 
               main="RUV Ctrl vs CRISPR PC-com", ylim=ylim); drawLines()
DESeq2::plotMA(ruv_res_crisp_CD34_pc_com,
               main="RUV CRISPR vs CD34 PC-com", ylim=ylim); drawLines()

# ENSG00000174059 is CD34 ENSG ID 
ruv_res_con_CD34_pc_com["ENSG00000174059",]
ruv_res_con_crisp_pc_com["ENSG00000174059",]
ruv_res_crisp_CD34_pc_com["ENSG00000174059",]

# looking at the 
summary(ruv_res_con_CD34_pc_com) # 123 up and 38 down
summary(ruv_res_con_crisp_pc_com) # 104 up and 22 down
summary(ruv_res_crisp_CD34_pc_com) # only CD34 up!

# grabbing the ENSG IDs from the differentially expressed genes 
ruv_res_con_CD34_pc_com_nona <- ruv_res_con_CD34_pc_com[!is.na(ruv_res_con_CD34_pc_com$padj),]
ruv_res_con_CD34_pc_com_IDs <-
  rownames(ruv_res_con_CD34_pc_com_nona)[ruv_res_con_CD34_pc_com_nona$padj<0.05]
length(ruv_res_con_CD34_pc_com_IDs) #161
num_CD34Vctrl <- length(ruv_res_con_CD34_pc_com_IDs) 

ruv_res_con_crisp_pc_com_IDs <-
  rownames(ruv_res_con_crisp_pc_com)[ruv_res_con_crisp_pc_com$padj<0.05]
length(ruv_res_con_crisp_pc_com_IDs) #125
num_crispVctrl <- length(ruv_res_con_crisp_pc_com_IDs) 

sum(ruv_res_con_CD34_pc_com_IDs %in% ruv_res_con_crisp_pc_com_IDs) # 97 are the same 
num_overlaps <- sum(ruv_res_con_CD34_pc_com_IDs %in% ruv_res_con_crisp_pc_com_IDs)

ruv_res_pc_com_IDs <- ruv_res_con_CD34_pc_com_IDs[(ruv_res_con_CD34_pc_com_IDs %in% ruv_res_con_crisp_pc_com_IDs)]
```

```{r com-DE-table, echo=TRUE, warning=FALSE, message = FALSE}

ruv_res_con_CD34_pc_com_sig <- as.data.frame(ruv_res_con_CD34_pc_com[rownames(ruv_res_con_CD34_pc_com) %in% ruv_res_pc_com_IDs,])
ruv_res_con_CD34_pc_com_sig$Ensembl_ID <- rownames(ruv_res_con_CD34_pc_com_sig)

ruv_res_con_crisp_pc_com_sig <- as.data.frame(ruv_res_con_crisp_pc_com[rownames(ruv_res_con_crisp_pc_com) %in% ruv_res_pc_com_IDs,])
ruv_res_con_crisp_pc_com_sig$Ensembl_ID <- rownames(ruv_res_con_crisp_pc_com_sig)

ruv_res_con_C34_CRISPR_pc_com_sig <- merge(x = ruv_res_con_CD34_pc_com_sig,
                                           y = ruv_res_con_crisp_pc_com_sig[,-1],
                                           by = "Ensembl_ID", all=T)
# write.table(ruv_res_con_C34_CRISPR_pc_com_sig, "ruv_res_con_C34_CRISPR_pc_com_sig2.txt",
#              row.names = F, col.names = TRUE, sep="\t", append=F, quote = F)
# write.csv(ruv_res_con_C34_CRISPR_pc_com_sig, "ruv_res_con_C34_CRISPR_pc_com_sig2.txt",
#              row.names = F, col.names = TRUE, quote = F)
```
Let's make an upset plot with the combined and the individual replicates. 

```{r upset-com-DE, echo=TRUE, warning=FALSE, message = FALSE, eval=TRUE}

# Setting up the dataframe for the upset plot. 
ruv_res_con_CD34_pc_com_IDs_upsetdf <- data.frame(Gene=ruv_res_con_CD34_pc_com_IDs,
                                                  Com_ctrl_v_CD34=1)
ruv_res_con_crisp_pc_com_IDs_upsetdf <- data.frame(Gene=ruv_res_con_crisp_pc_com_IDs,
                                                  Com_ctrl_v_CRISPR=1)
Combined_upsetdf <- merge(ruv_res_con_CD34_pc_com_IDs_upsetdf, 
                            ruv_res_con_crisp_pc_com_IDs_upsetdf,
                            by="Gene", all=T)
dim(Combined_upsetdf) # 189   3

Combined_upsetdf_nona <- replace(Combined_upsetdf,is.na(Combined_upsetdf),0)

sum(duplicated(Combined_upsetdf_nona$Gene))
upset(Combined_upsetdf_nona, nsets = 2, number.angles = 0, point.size = 3.5, 
      line.size = 1.5, mainbar.y.label = "DE Gene Intersections", 
      sets.x.label = "# of DE Genes", text.scale = c(1.3, 1, 1, 1, 1, 1), order.by = "freq")

# I would look at the gene ontology enrichment of the combined overlapping 97 genes. 
# saveRDS(ruv_res_pc_com_IDs, "ruv_res_pc_com_IDs2.rds")
# ruv_res_pc_com_IDs
# any(ruv_res_pc_com_IDs == "ENSG00000116721")
```

Making a diagram of the overlap between treatment comparisons
```{r com-VD, echo=TRUE, warning=FALSE, message = FALSE, eval=TRUE}
grid.newpage()
draw.pairwise.venn(num_crispVctrl, num_CD34Vctrl, num_overlaps, 
                   category = c("CRISPR v. Ctrl", "CD34 v. Ctrl"), 
                   lty = rep("blank", 2), fill = c("#7570B3", "#D95F02"),
                   alpha = rep(0.5, 2), cat.pos = c(0, 0), 
                   cat.dist = rep(0.025, 2))
```

```{r com-VD-write, echo=TRUE, warning=FALSE, message = FALSE, eval=FALSE}
pdf("Compare_VD_1.pdf", width=6, height = 4, family="ArialMT")
 draw.pairwise.venn(num_crispVctrl, num_CD34Vctrl, num_overlaps, 
                   category = c("CRISPR v. Ctrl", "CD34 v. Ctrl"), 
                   lty = rep("blank", 2), fill = c("#7570B3", "#D95F02"),
                   alpha = rep(0.5, 2), cat.pos = c(0, 0), 
                   cat.dist = rep(0.025, 2))
dev.off()
```

Using GeneOverlap to test the significance of overlap between the differentially expressed gene sets
```{r DEG-overlap, echo=TRUE, warning=FALSE, message = FALSE, eval=FALSE}

go.obj <- newGeneOverlap(ruv_res_con_CD34_pc_com_IDs,
                          ruv_res_con_crisp_pc_com_IDs,
                          genome.size=length(featureNames(eset_pc_norm_com)))
go.obj <- testGeneOverlap(go.obj)
print(go.obj)

```
Checking to make sure that the chosen qPCR validation targets are within the 97 genes overlapping between the comparisons. 

```{r qPCR-check, echo=TRUE, warning=FALSE, message = FALSE, eval=TRUE}
# in order: NFKB2, TRIB3, RelB, EGR1, JUNB, DDIT3, fosl1
qPCR_IDs <- c("ENSG00000077150", "ENSG00000101255", "ENSG00000104856", 
              "ENSG00000120738", "ENSG00000171223", "ENSG00000175197",
              "ENSG00000175592") 
length(qPCR_IDs) # 7

# check IDs
sum(ruv_res_pc_com_IDs %in% qPCR_IDs) #7
```

Making annotated MA plots using the ggplot package. First for CRISPR CD34 vs. Control

```{r MAplot-ctrl_CD34, echo=TRUE, warning=FALSE, message = FALSE, eval=TRUE}

head(ruv_res_con_CD34_pc_com)
color_genes <- c("#000000", "#FF0000", "#0432FF", "#FF40FF", "#548235", "#FF9300", "#942093", "#00FDFF")

# color CD34 red
ruv_res_con_CD34_pc_com$color <- 1
ruv_res_con_CD34_pc_com$color[rownames(ruv_res_con_CD34_pc_com)=="ENSG00000174059"] <- 2

# make non-DE transparent
ruv_res_con_CD34_pc_com$trans <- 0.1
ruv_res_con_CD34_pc_com$trans[which(ruv_res_con_CD34_pc_com$padj < 0.05)] <- 1 

# make CD34 larger
ruv_res_con_CD34_pc_com$size <- 1
ruv_res_con_CD34_pc_com$size[rownames(ruv_res_con_CD34_pc_com)=="ENSG00000174059"] <- 1.5

ruv_res_con_CD34_pc_com_df <- as.data.frame(ruv_res_con_CD34_pc_com)
head(ruv_res_con_CD34_pc_com_df)
str(ruv_res_con_CD34_pc_com_df)
ruv_res_con_CD34_pc_com_df$color <- factor(ruv_res_con_CD34_pc_com_df$color)
MAplot_Ctrl_v_CD34 <- ggplot(data = ruv_res_con_CD34_pc_com_df, aes(x = baseMean, y = log2FoldChange, color = color, alpha = trans, size=size)) +
  geom_point() +
  scale_x_continuous(trans = 'log10') +
  scale_alpha_continuous(guide=FALSE) +
  scale_size_continuous(guide=FALSE, range = c(1,3)) +
  scale_color_manual(values=c("black", "red"), guide=FALSE) +
  ylim(c(-7, 7)) +
  geom_hline(yintercept=c(-1,1), linetype="dotted") +
  xlab("mean expression") +
  ylab("log fold change") +
  ggtitle("Control vs. CD34") +
  theme_tufte()
MAplot_Ctrl_v_CD34
#pdf("MAplot_Ctrl_v_CD34.pdf", width=6, height = 4, family="ArialMT")
#MAplot_Ctrl_v_CD34
#dev.off()
```

Now for the Control vs. CRISPR plot

```{r MAplot-ctrl_CRISPR, echo=TRUE, warning=FALSE, message = FALSE, eval=TRUE}

# color CD34 red
ruv_res_con_crisp_pc_com$color <- 1
ruv_res_con_crisp_pc_com$color[rownames(ruv_res_con_crisp_pc_com)=="ENSG00000174059"] <- 2

# make non-DE transparent
ruv_res_con_crisp_pc_com$trans <- 0.1
ruv_res_con_crisp_pc_com$trans[which(ruv_res_con_crisp_pc_com$padj < 0.05)] <- 1 

# make CD34 larger
ruv_res_con_crisp_pc_com$size <- 1
ruv_res_con_crisp_pc_com$size[rownames(ruv_res_con_crisp_pc_com)=="ENSG00000174059"] <- 1.5

ruv_res_con_crisp_pc_com_df <- as.data.frame(ruv_res_con_crisp_pc_com)
head(ruv_res_con_crisp_pc_com_df)
str(ruv_res_con_crisp_pc_com_df)
ruv_res_con_crisp_pc_com_df$color <- factor(ruv_res_con_crisp_pc_com_df$color)

MAplot_Ctrl_v_CRISPR <- ggplot(data = ruv_res_con_crisp_pc_com_df, aes(x = baseMean, y = log2FoldChange, color = color, alpha = trans, size=size)) +
  geom_point() +
  scale_x_continuous(trans = 'log10') +
  scale_alpha_continuous(guide=FALSE) +
  scale_size_continuous(guide=FALSE, range = c(1,3)) +
  scale_color_manual(values=c("black", "red"), guide=FALSE) +
  ylim(c(-7, 7)) +
  geom_hline(yintercept=c(-1,1), linetype="dotted") +
  xlab("mean expression") +
  ylab("log fold change") +
  ggtitle("Control vs. CRISPR") +
  theme_tufte()
MAplot_Ctrl_v_CRISPR
#pdf("MAplot_Ctrl_v_CRISPR.pdf", width=6, height = 4, family="ArialMT")
#MAplot_Ctrl_v_CRISPR
#dev.off()

# remove CD34 highlight
ruv_res_con_crisp_pc_com$size[rownames(ruv_res_con_crisp_pc_com)=="ENSG00000174059"] <- 1
ruv_res_con_crisp_pc_com$color[rownames(ruv_res_con_crisp_pc_com)=="ENSG00000174059"] <- 1
ruv_res_con_crisp_pc_com_df <- as.data.frame(ruv_res_con_crisp_pc_com)
MAplot_Ctrl_v_CRISPR2 <- ggplot(data = ruv_res_con_crisp_pc_com_df, aes(x = baseMean, y = log2FoldChange, color = as.factor(color), alpha = trans)) +
  geom_point() +
  scale_x_continuous(trans = 'log10') +
  scale_alpha_continuous(guide=FALSE) +
  scale_color_manual(values=c("black", "red"), guide=FALSE) +
  ylim(c(-7, 7)) +
  geom_hline(yintercept=c(-1,1), linetype="dotted") +
  xlab("mean expression") +
  ylab("log fold change") +
  ggtitle("Control vs. CRISPR") +
  theme_tufte()
MAplot_Ctrl_v_CRISPR2
#pdf("MAplot_Ctrl_v_CRISPR2.pdf", width=6, height = 4, family="ArialMT")
#MAplot_Ctrl_v_CRISPR2
#dev.off()
```

CRISPR vs. CRISPR CD34 plot

```{r MAplot-CRSIPR_CD34, echo=TRUE, warning=FALSE, message = FALSE, eval=TRUE}

# color CD34 red
ruv_res_crisp_CD34_pc_com$color <- 1
ruv_res_crisp_CD34_pc_com$color[rownames(ruv_res_crisp_CD34_pc_com)=="ENSG00000174059"] <- 2

# make non-DE transparent
ruv_res_crisp_CD34_pc_com$trans <- 0.1
ruv_res_crisp_CD34_pc_com$trans[which(ruv_res_crisp_CD34_pc_com$padj < 0.05)] <- 1 

# make CD34 larger
ruv_res_crisp_CD34_pc_com$size <- 1
ruv_res_crisp_CD34_pc_com$size[rownames(ruv_res_crisp_CD34_pc_com)=="ENSG00000174059"] <- 1.5

ruv_res_crisp_CD34_pc_com_df <- as.data.frame(ruv_res_crisp_CD34_pc_com)
head(ruv_res_crisp_CD34_pc_com_df)
str(ruv_res_crisp_CD34_pc_com_df)
ruv_res_crisp_CD34_pc_com_df$color <- factor(ruv_res_crisp_CD34_pc_com_df$color)
MAplot_CRSIPR_v_CD34 <- ggplot(data = ruv_res_crisp_CD34_pc_com_df, aes(x = baseMean, y = log2FoldChange, color = color, alpha = trans, size=size)) +
  geom_point() +
  scale_x_continuous(trans = 'log10') +
  scale_alpha_continuous(guide=FALSE) +
  scale_size_continuous(guide=FALSE, range = c(1,3)) +
  scale_color_manual(values=c("black", "red"), guide=FALSE) +
  ylim(c(-7, 7)) +
  geom_hline(yintercept=c(-1,1), linetype="dotted") +
  xlab("mean expression") +
  ylab("log fold change") +
  ggtitle("CRISPR vs. CD34") +
  theme_tufte()
MAplot_CRSIPR_v_CD34
#pdf("MAplot_CRSIPR_v_CD34.pdf", width=6, height = 4, family="ArialMT")
#MAplot_CRSIPR_v_CD34
#dev.off()
```

Now to make MAplots highlighting the six genes tested by qPCR. 

First for Control vs. CRISPR

```{r MAplot-Ctrl-CRSIPR-qpcr, echo=TRUE, warning=FALSE, message = FALSE, eval=TRUE}

# color genes
ruv_res_con_crisp_pc_com$color <- 1
ruv_res_con_crisp_pc_com$color[rownames(ruv_res_con_crisp_pc_com)=="ENSG00000174059"] <- 2
ruv_res_con_crisp_pc_com$color[rownames(ruv_res_con_crisp_pc_com)=="ENSG00000104856"] <- 3
ruv_res_con_crisp_pc_com$color[rownames(ruv_res_con_crisp_pc_com)=="ENSG00000077150"] <- 4
ruv_res_con_crisp_pc_com$color[rownames(ruv_res_con_crisp_pc_com)=="ENSG00000175197"] <- 5
ruv_res_con_crisp_pc_com$color[rownames(ruv_res_con_crisp_pc_com)=="ENSG00000171223"] <- 6
ruv_res_con_crisp_pc_com$color[rownames(ruv_res_con_crisp_pc_com)=="ENSG00000175592"] <- 7
ruv_res_con_crisp_pc_com$color[rownames(ruv_res_con_crisp_pc_com)=="ENSG00000101255"] <- 8

# make non-DE transparent
ruv_res_con_crisp_pc_com$trans <- 0.1
ruv_res_con_crisp_pc_com$trans[which(ruv_res_con_crisp_pc_com$padj < 0.05)] <- 1 

# make CD34/genes larger
ruv_res_con_crisp_pc_com$size <- 1
GOI <- c("ENSG00000174059", "ENSG00000101255", "ENSG00000077150", "ENSG00000175197",
         "ENSG00000171223", "ENSG00000175592", "ENSG00000104856")

ruv_res_con_crisp_pc_com$size[rownames(ruv_res_con_crisp_pc_com) %in% GOI] <- 1.5

ruv_res_con_crisp_pc_com_df <- as.data.frame(ruv_res_con_crisp_pc_com)
head(ruv_res_con_crisp_pc_com_df)
str(ruv_res_con_crisp_pc_com_df)
ruv_res_con_crisp_pc_com_df$color <- factor(ruv_res_con_crisp_pc_com_df$color)
levels(ruv_res_con_crisp_pc_com_df$color)
MAplot_Ctrl_v_CRISPR_qPCR <- ggplot(data = ruv_res_con_crisp_pc_com_df, aes(x = baseMean, y = log2FoldChange, color = color, alpha = trans, size=size)) +
  geom_point() +
  scale_x_continuous(trans = 'log10') +
  scale_alpha_continuous(guide=FALSE) +
  scale_size_continuous(guide=FALSE, range = c(1,3)) +
  scale_color_manual(values=color_genes, guide=FALSE) +
  ylim(c(-7, 7)) +
  geom_hline(yintercept=c(-1,1), linetype="dotted") +
  xlab("mean expression") +
  ylab("log fold change") +
  ggtitle("Control vs. CRISPR") +
  theme_tufte()
MAplot_Ctrl_v_CRISPR_qPCR
#pdf("MAplot_Ctrl_v_CRISPR_qPCR.pdf", width=6, height = 4, family="ArialMT")
#MAplot_Ctrl_v_CRISPR_qPCR
#dev.off()
```

Now for Control vs. CRISPR CD34

```{r MAplot-Ctrl-CD34-qPCR, echo=TRUE, warning=FALSE, message = FALSE, eval=TRUE}

# color genes
ruv_res_con_CD34_pc_com$color <- 1
ruv_res_con_CD34_pc_com$color[rownames(ruv_res_con_CD34_pc_com)=="ENSG00000174059"] <- 2
ruv_res_con_CD34_pc_com$color[rownames(ruv_res_con_CD34_pc_com)=="ENSG00000104856"] <- 3
ruv_res_con_CD34_pc_com$color[rownames(ruv_res_con_CD34_pc_com)=="ENSG00000077150"] <- 4
ruv_res_con_CD34_pc_com$color[rownames(ruv_res_con_CD34_pc_com)=="ENSG00000175197"] <- 5
ruv_res_con_CD34_pc_com$color[rownames(ruv_res_con_CD34_pc_com)=="ENSG00000171223"] <- 6
ruv_res_con_CD34_pc_com$color[rownames(ruv_res_con_CD34_pc_com)=="ENSG00000175592"] <- 7
ruv_res_con_CD34_pc_com$color[rownames(ruv_res_con_CD34_pc_com)=="ENSG00000101255"] <- 8

# make non-DE transparent
ruv_res_con_CD34_pc_com$trans <- 0.1
ruv_res_con_CD34_pc_com$trans[which(ruv_res_con_CD34_pc_com$padj < 0.05)] <- 1 

# make CD34/genes larger
ruv_res_con_CD34_pc_com$size <- 1
GOI <- c("ENSG00000174059", "ENSG00000101255", "ENSG00000077150", "ENSG00000175197",
         "ENSG00000171223", "ENSG00000175592", "ENSG00000104856")

ruv_res_con_CD34_pc_com$size[rownames(ruv_res_con_CD34_pc_com) %in% GOI] <- 1.5

ruv_res_con_CD34_pc_com_df <- as.data.frame(ruv_res_con_CD34_pc_com)
head(ruv_res_con_CD34_pc_com_df)
str(ruv_res_con_CD34_pc_com_df)
ruv_res_con_CD34_pc_com_df$color <- factor(ruv_res_con_CD34_pc_com_df$color)
levels(ruv_res_con_CD34_pc_com_df$color)
MAplot_Ctrl_v_CD34_qPCR <- ggplot(data = ruv_res_con_CD34_pc_com_df, aes(x = baseMean, y = log2FoldChange, color = color, alpha = trans, size=size)) +
  geom_point() +
  scale_x_continuous(trans = 'log10') +
  scale_alpha_continuous(guide=FALSE) +
  scale_size_continuous(guide=FALSE, range = c(1,3)) +
  scale_color_manual(values=color_genes, guide=FALSE) +
  ylim(c(-7, 7)) +
  geom_hline(yintercept=c(-1,1), linetype="dotted") +
  xlab("mean expression") +
  ylab("log fold change") +
  ggtitle("Control vs. CD34") +
  theme_tufte()
MAplot_Ctrl_v_CD34_qPCR
#pdf("MAplot_Ctrl_v_CD34_qPCR.pdf", width=6, height = 4, family="ArialMT")
#MAplot_Ctrl_v_CD34_qPCR
#dev.off()
```


Outputting the Session Info
```{r sessionInfo, echo=TRUE, warning=FALSE, message = FALSE, eval=TRUE}
sessionInfo()
```